engine/src/main/java/com/datatorrent/stram/engine/OperatorContext.java

```diff
@@ -84,12 +87,13 @@ public void setIdleTimeout(long idleTimeout)
    * @param attributes the value of attributes
    * @param parentContext
    */
-  public OperatorContext(int id, AttributeMap attributes, Context parentContext)
+  public OperatorContext(int id, String operatorName, AttributeMap attributes, Context parentContext)
```

Comment:
id, name and attributes are public fields on OperatorDeployInfo. Using OperatorDeployInfo will allow avoiding changes to the OperatorContext constructor each time a new field is introduced. There seems to be the dependency in production code between OperatorContext and OperatorDeployInfo anyway.